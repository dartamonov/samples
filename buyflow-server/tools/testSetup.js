// Tests are placed alongside files under test.
// This file does the following:
// 1. Sets the environment to 'production' so that
//    dev-specific babel config in .babelrc doesn't run.
// 2. Disables Webpack-specific features that Mocha doesn't understand.
// 3. Registers babel for transpiling our code for testing.

// This assures the .babelrc dev config (which includes
// hot module reloading code) doesn't apply for tests.
// process.env.NODE_ENV = 'production';
// process.env.NODE_ENV = 'test';

// Register babel so that it will transpile ES6 to ES5
// before our tests run.
require('babel-register')();
//require("babel-polyfill");
require("babel-jest");
