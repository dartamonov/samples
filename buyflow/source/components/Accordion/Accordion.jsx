import React, { Component, PropTypes } from 'react';
import classNames from 'classnames';

class Accordion extends Component {

  animate(e) {
    e.preventDefault();
    $(this.refs.accordion).find('.accordion__body').slideToggle( "slow", function() {
      // animation complete
    });
  }

  render() {
    const {
      active = false,
      children,
      className
    } = this.props;

    return (
      <div ref="accordion" className={classNames(className, 'accordion', { 'is-active': active })} aria-expanded={active}>
        {children}
      </div>
    );
  }
}

Accordion.propTypes = {
  active: PropTypes.bool,
  children: PropTypes.object,
  className: PropTypes.string
};

export default Accordion;
