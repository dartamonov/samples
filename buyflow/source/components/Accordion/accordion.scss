.accordion {
  margin-bottom: 20px;

  &.is-active {
    .accordion__body {
      display: block;
    }
  }
}

.accordion__title {
  margin-bottom: 2px;
}

.accordion__body {
  display: none;
  .hg-panel{
    background-color: $coolGrey1;
  }
}

.accordion__trigger {}
