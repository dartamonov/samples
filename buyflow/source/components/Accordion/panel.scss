.hg-panel {
  background: #FFF;
  padding: unit-calc(20);
  position: relative;

  &--secondary {
    background: $coolGrey1;
  }

  hr {
    margin-top: 0;
    background: $coolGrey5;
  }
}

.hg-panel__content {
  > :last-child {
    margin-bottom: 0;
  }
}

.hg-panel--action-right-at-768 {
  .hg-panel__action {
    @include media-breakpoint-up('lg') {
      @include vertical-align;
      right: 12px;
    }
  }
}

.hg-panel--transparent {
  background: transparent;
  padding: 0 0 20px;

  @include media-breakpoint-up('lg') {
    padding: 20px;
  }
}

.hg-panel--bordered {
  border: 1px solid $grey9;
}

.hg-panel--bordered-no-top {
  border-top: 0;
}

.hg-panel--border-top {
  border-top: 1px solid $grey9;
}

.hg-panel--cool-grey1 {
  background: $coolGrey1;
}

.hg-panel-pipe-container {
  border-bottom: 1px solid $grey9;
  @include media-breakpoint-up('lg') {
    border-bottom: 0;
    border-right: 1px solid $grey9;
    margin-right: unit-calc(-20px);
    padding-right: unit-calc(20px);
  }

  .x-input--radio {
    align-items: flex-start;
    &:before {
      margin-top: unit-calc(3px);
    }
  }
}

.hg-panel--block {
  display: block;
  flex: 1;
}

.hg-panel--collapse-top {
  padding-top: 0;
  border-top: 0;
}

.hg-panel--slim {
  padding: unit-calc(12 20);
}

.hg-panel--cool-greyN {
  background-color: #F9FAFA;
}
