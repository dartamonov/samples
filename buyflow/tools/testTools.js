import renderer from 'react-test-renderer';

export const takeSnapshot = x => renderer.create(x).toJSON();