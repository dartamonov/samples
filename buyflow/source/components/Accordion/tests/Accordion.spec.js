import React from 'react';
import Accordion from '../Accordion.jsx';
import {takeSnapshot} from '../../../tools/testTools';

describe('Accordion component', () => {
  it('handles active accordion ', () => {
    const snapshot = takeSnapshot(
      <Accordion active={true}>
        <div>Content</div>
      </Accordion>
    );

    expect(snapshot).toMatchSnapshot();
  });

  it('handles inactive accordion ', () => {
    const snapshot = takeSnapshot(
      <Accordion active={false}>
        <div>Content</div>
      </Accordion>
    );

    expect(snapshot).toMatchSnapshot();
  });
});
