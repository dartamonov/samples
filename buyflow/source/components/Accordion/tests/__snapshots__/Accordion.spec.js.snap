// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Accordion component handles active accordion  1`] = `
<div
  aria-expanded={true}
  className="accordion is-active"
>
  <div>
    Content
  </div>
</div>
`;

exports[`Accordion component handles inactive accordion  1`] = `
<div
  aria-expanded={false}
  className="accordion"
>
  <div>
    Content
  </div>
</div>
`;
